src/Shouldly.Tests/ShouldBeLessThanOrEqualTo/CustomObjectScenario.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Shouldly.Tests.Strings;
+using Shouldly.Tests.TestHelpers;
+using Xunit;
+
+namespace Shouldly.Tests.ShouldBeLessThanOrEqualTo
+{
+    public class CustomObjectScenario
+    {
+        [Fact]
+        public void CustomObjectScenarioShouldFail()
+        {
+            var customA = new Custom { Val = 2 };
+            var customB = new Custom { Val = 1 };
+            var comparer = new CustomComparer<Custom>();
+            Verify.ShouldFail(() =>
+customA.ShouldBeLessThanOrEqualTo(customB, comparer, "Some additional context"),
+
+errorWithSource:
+@"customA
+    should be less than or equal to
+Shouldly.Tests.TestHelpers.Custom (000000)
+    but was
+Shouldly.Tests.TestHelpers.Custom (000000)
+
```

Comment:
It would be good to output the value of customA and customB, I propose an error message like:

**With source:**
"customA (2) should be less than or equal to customB (1) but was not"

**Without source**
"2 should be less than or equal to 1 but was not"

What are your thoughts @JakeGinnivan ?